packages/components/src/toggle-group-control/toggle-group-control-option-icon/component.tsx

```diff
@@ -0,0 +1,55 @@
+/**
+ * WordPress dependencies
+ */
+import { Icon } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import type { WordPressComponentProps } from '../../ui/context';
+import type { ToggleGroupControlOptionIconProps } from '../types';
+import { ToggleGroupControlOptionBase } from '../toggle-group-control-option-base';
+
+export default function ToggleGroupControlOptionIcon(
+	props: WordPressComponentProps<
+		ToggleGroupControlOptionIconProps,
+		'button'
+	>
```

Comment:
We should mark `ToggleGroupControlOptionBase`, `ToggleGroupControlOption` and `ToggleGroupControlOptionIcon` as non-polymorphic, since they are internally always going to render `as="button"`

```suggestion
	props: WordPressComponentProps<
		ToggleGroupControlOptionIconProps,
		'button',
		false
	>
```

Reply:
Why we didn't have this before?

Reply:
Not sure. From a quick look, it seems that the component's prop were already like this since the component was called `SegmentedControl`. Probably just something that got overlooked?